schemas/session-replay/metadata-record-schema.json

```diff
@@ -0,0 +1,60 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "metadata-record-schema.json",
+    "title": "MetadataRecord",
+    "type": "object",
+    "description": "Schema of a Record which contains the screen properties.",
+    "allOf": [
+        {
+            "$ref": "_common-record-schema.json"
+        },
+        {
+            "required": [
+                "type",
+                "source",
+                "data"
+            ],
+            "properties": {
+                "type": {
+                    "type": "integer",
+                    "description": "The type of this Record.",
+                    "const": 4,
+                    "readOnly": true
+                },
+                "source": {
+                    "type": "string",
+                    "description": "The source of this record",
+                    "enum": [
+                        "android",
+                        "ios",
+                        "browser",
+                        "flutter",
+                        "react-native"
+                    ],
+                    "readOnly": true
+                },
```

Comment:
**suggestion:** move this to data and make it mandatory

Reply:
**question:** what about `device_type`?

Reply:
Let's leave this for later.

Reply:
> Actually on the first point...I am not sure this should be at the `data` level. The `source` is a different topic than the actual data. What do you think ?


No the source is the data of the meta record. the only outside of data is the timestamp

> Let's leave this for later.

😢

Reply:
Ok so all things considered I think it's nice to have the source field mandatory for all the record types, not only the `MetadataRecord`.

On our side it could allow us to keep the players agnostic and leave the `if mobile then` to the sandbox and if we know the source for each record it will simply the implementation

Reply:
that means this will be a part of the `_common-record-schema.json`. Do we all agree here ? @ThibautGeriz @mquentin @ncreated @acorretti ?

Reply:
I would rather leave them on the meta record honestly, a small state is better than duplicating the field everywhere. We would only need it on the fullsnapshot & incremental. So if we want to avoid state, we should only add it to those. But I am not super fan of this.

I am curious to have your opinion @acorretti & @BenoitZugmeyer

Reply:
> have the source field mandatory for all the record types

This would be against [the guidelines](https://github.com/DataDog/rum-events-format/#guidelines): since older versions of the Browser SDK would not send the `source` field, you cannot assume that it will be always defined. Beside that, we should avoid adding 10+ bytes to every records.

What about moving this information to the Segment instead?
* no need to maintain a state from the last Metadata record
* we would be able to know the source even if there is no Metadata record in a segment (might be useful when debugging)
* it could be indexed (not sure if useful but still)

Reply:
actually, Benoit shared his opinion here https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2471460887/RFC+-+Data+Format+for+Mobile+Session+Replay?focusedCommentId=2515730503

> What would be the use-case? I think that it would be more convenient to have it as a segment meta property than in a record, because to won’t need to find the Meta record inside a Segment to access the information. Also, Segments don’t always have Meta records.
> Also I have the secret ambition to remove the Meta record one day. URL is useless, and for the viewport size we have the visualviewport record.


I see three options. Am I missing something?


### 1. Add it in the segment metadata

Add it in the segment metadata. Once received, the fetcher will add to the `RecordWithViewId` data-structure and the renderer will have have it with all records

**Pros:**
- have the info on all the segments (we don't need it right now)
- no duplication at the sdk level
- no need to keep a state in the HTML renderer
- Benoit can possibly complete his secret dream


**Cons:**
- need to add logic in the player itself (no big deal)
- internal message a little big bigger (very small consequence)
- the initialisation state is split between the meta record & segment itself
  - what about if we want to add the device_type if we were to add it, where should it go?

### 2. Add it in the meta record

The meta record create the sandbox and keep the source as a react state

**Pros:**
- we already have all the initial state in there to create a sandbox like the URL (used for the base tag) or the initial dimension
- no duplication at the sdk level
- simple

**Cons:**
- we need to add a new state HTML renderer
- Benoit cannot complete his secret dream 😿 


### 3. Add it in all records

**Pros:**
- have the info on all the record (we don't need it right now)
- no need to keep a state in the HTML renderer
- Benoit can possibly complete his secret dream


**Cons:**
- duplication at the sdk level
- internal message a little big bigger (very small consequence)
- the initialisation state is split between the meta record & inside all records
  - what about if we want to add the device_type, where should it go?

I would choose 1 or 2

Reply:
so we add it in the `MetadataRecord#data#source` ?

Reply:
Indeed I get it,
either we go with the first or second option. I like the first one but we already maintain a state built on the meta record so even the second option works for me.

anyway, in both case with will provide the `isMobile` context to the sandbox iframe using the message bridge.